todo_list_app/views.py

```diff
@@ -130,110 +130,29 @@ def perform_create(self, serializer):
         serializer.save(user=self.request.user)
 
 
-@login_required
-def tasks(request):
-    query = request.GET.get('q')
-    try:
-        task_list = Task.objects.filter(category__user=request.user)
+class TaskViewSet(viewsets.ModelViewSet):
+    serializer_class = TaskSerializer
+    permission_classes = [IsAuthenticated, IsOwnerTaskPermission]
+
+    def get_queryset(self):
+        query = self.request.query_params.get('q')
+        task_list = Task.objects.filter(category__user=self.request.user)
 
         if query:
             task_list = task_list.filter(
                 Q(title__icontains=query) |
                 Q(description__icontains=query) |
                 Q(category__name__icontains=query)
```

Comment:
What is the exact purpose of `icontaints` and specifically `i` in it?

Reply:
contains is used to perform case-sensitive search based on the value and icontains perform case-insensitive search